### PR TITLE
Moved self-test from Bio\PDB\MMCIFParser.py

### PR DESCRIPTION
**Pull request by Gasta88**
As referenced in issue #820, I've moved the self-test from the cited file to a new one: `Tests\test_PDB_MMCIFParser.py`.

### CODE REVIEW

File: Tests/test_PDB-MMCIFParser.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Copyright 2017 by Francesco Gastaldello. All rights reserved.
+# Revisions copyright 2017 by Peter Cock.  All rights reserved.
```
Comment (peterjc) [inline Tests/test_PDB-MMCIFParser.py]: I haven't touched this file (yet).

File: Tests/test_PDB-MMCIFParser.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+#
+# Converted by Francesco Gastaldello from an older unit test copyright 2002
```
Comment (peterjc) [inline Tests/test_PDB-MMCIFParser.py]: The new test looks nothing like the original self-test, so I don't think you need this note?

**Comment by peterjc** [comment]
Can you rename the new file from ``test_PDB-MMCIFParser.py`` to ``test_PDB_MMCIFParser.py`` (with an underscore to match existing usage) please?

File: Tests/test_PDB_MMCIFParser.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+except ImportError:
+    from Bio import MissingPythonDependencyError
+    raise MissingPythonDependencyError(
+        "Install NumPy if you want to use Bio.PDB.")
+
+from Bio.PDB import MMCIFParser, FastMMCIFParser
+
+
+class MMCIFParserTests(unittest.TestCase):
+    r"""Test module MMCIFParser with CIF files located in TEst\PDB."""
```
Comment (peterjc) [inline Tests/test_PDB_MMCIFParser.py]: There's a case-oddity in the path, but you could just simplify this to:

```python
class MMCIFParserTests(unittest.TestCase):
    """Tests for MMCIFParser."""
```

File: Tests/test_PDB_MMCIFParser.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        """Test method MMCIFParser when reading a CIF file."""
+        filename = "PDB/1A8O.cif"
+        p = MMCIFParser()
+        structure = p.get_structure("test", filename)
+        res_list = [res.get_resname() for res in structure.get_residues()]
+        self.assertEqual(len(res_list), 158)
+        self.assertEqual("-".join(res_list), 'MSE-ASP-ILE-ARG-GLN-GLY-PRO-LYS-GLU-PRO-PHE-ARG-ASP-TYR-VAL-ASP-ARG-PHE-TYR-LYS-THR-LEU-ARG-ALA-GLU-GLN-ALA-SER-GLN-GLU-VAL-LYS-ASN-TRP-MSE-THR-GLU-THR-LEU-LEU-VAL-GLN-ASN-ALA-ASN-PRO-ASP-CYS-LYS-THR-ILE-LEU-LYS-ALA-LEU-GLY-PRO-GLY-ALA-THR-LEU-GLU-GLU-MSE-MSE-THR-ALA-CYS-GLN-GLY-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH-HOH')
+
+    def test_FastMMCIFParser(self):
+        """Test alternative version of the previous method when reading a CIF file."""
```
Comment (peterjc) [inline Tests/test_PDB_MMCIFParser.py]: Missing key word ``FastMMCIFParser`` here, and avoid words like "previous" as the test running framework may not run them in the order listed in the file.

**Comment by peterjc** [comment]
I was wondering why there was no meaningful change to the coverage - it looks like ``test_MMCIF.py`` already covers everything here. Sorry, I should have spotted this earlier.

Could you look at the existing ``test_MMCIF.py`` code, and if you agree your pull request can be simplified to just removing the self-test from ``Bio/PDB/MMCIFParser.py``

**Comment by Gasta88** [comment]
My bad, I should have check that at the beginning.

Do you think that the file should be renamed `test_PDB_MMCIFParser.py` since we are opting for this nomenclature with tests related to the `Bio.PDB` module?

**Comment by peterjc** [comment]
@lennax do you have thoughts on renaming ``test_MMCIF.py`` to ``test_PDB_MMCIFParser.py``?

**Comment by lennax** [comment]
@peterjc I am fine with renaming - it's definitely more clear to have the test filename match the module structure and name.

**Comment by peterjc** [comment]
Thanks @lennax.

@Gasta88 since you suggested it you could do the rename (as a separate pull request please after we've closed this one).

For this pull request just remove your ``test_PDB_MMCIFParser.py`` file, and it should be ready to merge 👍

**Comment by peterjc** [comment]
Thank you.

**Pull request merged**